src/app/hooks.ts

```diff
@@ -0,0 +1,5 @@
+import { AppDispatch, RootState } from './store';
+import { useDispatch, useSelector } from 'react-redux';
+
+export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
```

Comment:
The `useDispatch.withTypes<AppDispatch>()` syntax is incorrect. To create a typed `useDispatch` hook, you should use `() => useDispatch<AppDispatch>()`.